src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -29,22 +29,23 @@ namespace Microsoft.CodeAnalysis.Host;
 internal sealed partial class TemporaryStorageService : ITemporaryStorageServiceInternal
 {
     /// <summary>
-    /// The maximum size in bytes of a single storage unit in a memory mapped file which is shared with other
-    /// storage units.
+    /// The maximum size in bytes of a single storage unit in a memory mapped file which is shared with other storage
+    /// units.
     /// </summary>
     /// <remarks>
-    /// <para>This value was arbitrarily chosen and appears to work well. Can be changed if data suggests
-    /// something better.</para>
+    /// <para>The value of 256k reduced the number of files dumped to separate memory mapped files by 60% compared to
+    /// the next lower power-of-2 size for Roslyn.sln itself.</para>
     /// </remarks>
     /// <seealso cref="_weakFileReference"/>
-    private const long SingleFileThreshold = 128 * 1024;
+    private const long SingleFileThreshold = 256 * 1024;
 
     /// <summary>
     /// The size in bytes of a memory mapped file created to store multiple temporary objects.
     /// </summary>
     /// <remarks>
-    /// <para>This value was arbitrarily chosen and appears to work well. Can be changed if data suggests
-    /// something better.</para>
+    /// <para>This value (8mb) creates roughly 35 memory mapped files (around 300MB) to store the contents of all of
```

Comment:
>roughly 35 [](http://example.com/codeflow?start=39&length=10)

This isn't the 32 in the multiplation below? 

Knowing nothing about this code, is there a reason that the MultiFileBlockSize is increasing? Could it still be the same by changing the 32 to a 16 since you doubled SingleFileThreshold?

Reply:
It is not the 32.  Each large block we allocate is currently 4mb. This doubles that to halve the number of total Mmfs we need for Roslyn itself

Reply:
> Knowing nothing about this code, is there a reason that the MultiFileBlockSize is increasing? 

Yes. It's basically saying: we'd like each large block to hold at least 32 large files (or a lot more of we're question working with small files.  

It's a balance of not wanting to preallocate something huge, or something that grows is massive chunks (like, say 256 mb at a time), while also keeping it handle count low and not creating an excessive number of actual memory mapped files. 

These numbers strike a good balance. Growing 8mb at a time is fine, and we can still pack most files into those chunks

Reply:
There are two additional risks here:

1. We use bump-pointer allocation to fit new data into these blocks. The larger the item allowed, the larger the potential gap of unused space is at the end of a segment. By doubling the block size at the same time as doubling the threshold, I'm hoping these balance out.
2. We do not use compaction as part of our "GC strategy" for these blocks. If any single segment within the block is retained, the entire block will not be eligible for GC. If after running VS for a long period of time, you observe a significant number of MMF blocks of size 128KiB < size < 256KiB, this would indicate a potential problem with this change because many of those blocks could have increased in size to 8MiB (by being the last segment to retain a block).

Reply:
> We use bump-pointer allocation to fit new data into these blocks. The larger the item allowed, the larger the potential gap of unused space is at the end of a segment. By doubling the block size at the same time as doubling the threshold, I'm hoping these balance out.

I didn't see any problematic gaps measuring this. And, as you said, both are doubled.  So the expected %-loss should stay the same.

> Prior to this change, these segments would only end up retaining 4MiB, but this change would cause them to retain 8MiB.

Both of thse are so small as to be irrelevant afaict.  There is no realistic usage pattern i can think of either that would lead to excessive patterns being a problem with teh new numbers that wouldn't have already been a problem with the old ones.